src/adapter/pyth/PythOracle.sol

```diff
@@ -65,21 +94,43 @@ contract PythOracle is BaseAdapter {
         bool inverse = ScaleUtils.getDirectionOrRevert(_base, base, _quote, quote);
 
         PythStructs.Price memory priceStruct = _fetchPriceStruct();
+
         uint256 price = uint256(uint64(priceStruct.price));
+        int8 feedExponent = int8(baseDecimals) - int8(priceStruct.expo);
 
-        // priceStruct.expo will always be negative
-        uint8 feedExponent = uint8(int8(baseDecimals) - int8(priceStruct.expo));
-        Scale scale = ScaleUtils.from(quoteDecimals, feedExponent);
+        Scale scale;
+        if (feedExponent > 0) {
+            scale = ScaleUtils.from(quoteDecimals, uint8(feedExponent));
+        } else {
+            scale = ScaleUtils.from(quoteDecimals + uint8(-feedExponent), 0);
+        }
         return ScaleUtils.calcOutAmount(inAmount, price, scale, inverse);
     }
 
     /// @notice Get the latest Pyth price and perform sanity checks.
-    /// @dev Reverts conditions: price is negative, confidence interval is too wide,
-    /// exponent is positive, exponent
-    /// confidence is too wide, or exponent is too large.
+    /// @dev Revert conditions: update timestamp is too stale or too ahead, price is negative or zero,
+    /// confidence interval is too wide, exponent is positive or too small.
```

Comment:
`MAX_EXPONENT = 12` hence it can be positive. IMO the comment should say: `exponent is too large or too small`